framework/src/preconditioners/PhysicsBasedPreconditioner.C

```diff
@@ -138,11 +138,14 @@ PhysicsBasedPreconditioner::PhysicsBasedPreconditioner(const InputParameters & p
   for (unsigned int var = 0; var < n_vars; var++)
     addSystem(var, off_diag[var], _pre_type[var]);
 
-  // We don't want to be computing the big Jacobian!
-  _nl.nonlinearSolver()->jacobian = NULL;
   _nl.nonlinearSolver()->attach_preconditioner(this);
 
-  _fe_problem.solverParams()._type = Moose::ST_JFNK;
+  if (_fe_problem.solverParams()._type != Moose::ST_JFNK)
+  {
+    mooseWarning(
+        "PBP should be used with JFNK solve type, thus Executioner/solve_type is set to JFNK");
```

Comment:
I think this should be an error. People should fix their input files. Say something like "PBP can be used only with JFNK. Please, change your solve_type."